src/pudl/transform/classes.py

```diff
@@ -463,9 +502,8 @@ def distinct_domains(cls, params):
         * For all unit conversions calculate the range of original values that result
           from the inverse of the specified unit conversion applied to the valid
           ranges of values.
-
         * For all pairs of unit conversions verify that their original data ranges do
-          not overlap.
+          not overlap. Note that we also need to compare the unit conversion to itself.
```

Comment:
i don't really know what this means

Reply:
I think you're talking about the self-comparison here so I clarified that part.

Reply:
I was! I appreciate the addition. it does make it more clear but now i think it should maybe be moved into a separate bullet (?) and I think I don't fully understand what you are describing.

after reading through the code itself + the docs let me parrot back what I think is happening and you can correct me if i'm wrong (and hopefully understand where the docs aren't clear).

For ever `data_col`/categorical(`cat_col`/`cat_value`) combo, you can have several unit conversions. If i'm reading `UnitConversion` correctly, this refers to a series which has `from_unit` in the series name. So in theory, we could apply a list of unit conversions on the same series with the same original name? I'm having a hard time imagining why that is necessary... but if it is than it is!

So this validator is doing is doing is:
* we have a valid output range and a conversion, so we can construct what the range of potential input values will be result in a valid output. (note: using the language of "original values" here feels confusing to me... there is no input data yet) 
* Check whether the potential input values from each conversion overlap with any other conversion.
* Check whether the valid output ranges of each conversion overlap with any potential input value from other conversions.

I'm not sure if this is exactly correct but it took me a while to compile this (even maybe wrong!) understanding.

Reply:
I've expanded the explanation and added a simple example.